development/data-flows-for-analytics.md

```diff
@@ -0,0 +1,85 @@
+
+Data in the views and functions mentioned in this section is as accurate as the accuracy of the SQL queries. Best practice is to begin the process of defining these objects at design in order to align analytics and dashboards requirements with workflows being deployed.
+
+#### 3. Data use
+
+The data in PostgreSQL is mostly either used by direct querying or via dashboard visualizations. Database visualizations are built scoped to the requirements of supporting a successful deployment. In-depth data analysis is supported at the PostgreSQL level through updated contactviews, formviews, useviews and functions with access to these provided to relevant parties as and when needed.
+
+As mentioned previously, formviews are built to present data in a structure similar to the data collection tool (form) used. Useviews are tailored to align with a use case, mostly using the formviews are the data sources. These are fundamentally guided by design of the workflows and should be interpreted in the context of the design materials including a document explaining the definitions of variables used.
+
+The objects present here are not limited to views and functions. Additional tables can be added, for example providing mappings or supporting operations external to the functions available in the toolkit. In short, there is no limitation to the utility that can be added this level to support analytics and dashboards. That said, measures are taken to ensure controlled access, reliability and timely access of the data by the various parties. Some of these measures are:
+
+- roles and users allocation and deallocation done by specific roles within partner technical teams with support from Medic Mobile as needed;
+- access control management is left to the partner technical teams where possible;
+- dashboard data source refresh intervals set to align with project needs;
+- update of the data sources monitored to ensure updating works as expected;
+- review of the dashboards as part of the design process;
+- qualitative design activities to interrogate trends observed in the dashboards and iterate on them if need be;
+
+### Beyond current pipeline
+
+[cht-core](https://github.com/medic/cht-core) is mostly data collection tools and is the first in the data management pipeline. It is the core part of a deployment but the rest of the tools can be easily replaced with other preferred options. It also helps that couch2pg is an open source tool which provides the opportunity for collaboration to extend its functionality to support other implementations. Klipfolio, the tool that we currently use for visualizations, is a proprietary tool but there exists open source options such as [Apach Superset](https://superset.incubator.apache.org/) that are worth exploring.
+
+## Backup
+
+The machines running each of CouchDB and PostgreSQL instances are backed up daily.
+
+## Summary
+
+The diagram below summarizes the details in the document.
```

Comment:
I would actually probably leave this diagram out (ie remove it). It's pretty specific to our usage, includes very custom components that were manually created (dwh_impact, for example), and represents a structure that we are actively looking to replace.

Reply:
Done.